src/schema/v2/artist/index.ts

```diff
@@ -689,13 +697,13 @@ const Artist: GraphQLFieldConfig<void, ResolverContext> = {
       type: new GraphQLNonNull(GraphQLString),
     },
   },
-  resolve: (_root, { id }, { artistLoader }) => {
-    if (id.length === 0) {
-      return null
-    }
-    return artistLoader(id)
+  resolve: async (_root, { id }, { artistLoader }) => {
+    if (id.length === 0) return null
+    const artist = await artistLoader(id)
+    return cast(GravityArtistCodec, artist)
```

Comment:
Would it make more sense to do the casting int the loader itself?

Reply:
Yeah, definitely — good point.

Reply:
Maybe something in here: https://github.com/artsy/metaphysics/blob/master/src/lib/loaders/api/loader_interface.ts

Reply:
+1 to putting it in the loader. An artist type can be 'instantiated' from fetching (and resolving) the public JSON of that artist, as in this resolver, but can also be resolved from the short JSON in a list context.

In that case this resolver is skipped so that's not actually an issue, but it definitely feels better to bind with the loader (which is itself bound to the specific endpoint).